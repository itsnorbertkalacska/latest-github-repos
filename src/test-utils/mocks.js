export const rawRepo = {
  id: 1,
  name: "My Repo",
  description: "This is the description",
  stargazersCount: 111,
  htmlUrl: "github.com/my-repo",
};

const rawRepoB = {
  id: 2,
  name: "Second Repo",
  description: "Lorem Ipsum",
  stargazersCount: 2450,
  htmlUrl: "github.com/this-is-second",
};

export const repo = {
  ...rawRepo,
  isFavourited: false,
};

const repoB = {
  ...rawRepoB,
  isFavourited: true,
};

export const rawRepos = [rawRepo, rawRepoB];

export const repos = [repo, repoB];
