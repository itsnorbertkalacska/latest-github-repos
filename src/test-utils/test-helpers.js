import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";

import { theme } from "../theme";

export const renderWithProviders = (ui) => {
  function Wrapper({ children }) {
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
  }
  return render(ui, { wrapper: Wrapper });
};
