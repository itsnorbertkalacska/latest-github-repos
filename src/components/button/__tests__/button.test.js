import { fireEvent, screen } from "@testing-library/react";

import { renderWithProviders } from "../../../test-utils/test-helpers";
import { Button } from "../index";

describe("Button", () => {
  it("calls onClick on click", () => {
    const onClick = jest.fn();
    renderWithProviders(<Button onClick={onClick}>Click me</Button>);

    fireEvent.click(screen.getByText("Click me"));

    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it("passes props propery", () => {
    const { container } = renderWithProviders(
      <Button type="submit">Click me</Button>
    );
    // eslint-disable-next-line testing-library/no-node-access
    expect(container.firstChild).toMatchSnapshot();
  });
});
