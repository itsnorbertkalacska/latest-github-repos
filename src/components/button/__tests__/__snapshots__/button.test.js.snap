// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button passes props propery 1`] = `
<button
  class="sc-bczRLJ iulNEE"
  type="submit"
>
  Click me
</button>
`;
